thriftbp/prometheus.go

```diff
@@ -238,25 +275,52 @@ var (
 type clientPoolGaugeExporter struct {
 	slug string
 	pool clientpool.Pool
+
+	activeConnections prometheusbpint.HighWatermarkValue
 }
 
-func (e clientPoolGaugeExporter) Describe(ch chan<- *prometheus.Desc) {
+func (e *clientPoolGaugeExporter) Describe(ch chan<- *prometheus.Desc) {
 	// All metrics are described dynamically.
 }
 
-func (e clientPoolGaugeExporter) Collect(ch chan<- prometheus.Metric) {
+func (e *clientPoolGaugeExporter) Collect(ch chan<- prometheus.Metric) {
+	e.activeConnections.Set(int64(e.pool.NumActiveClients()))
+	active, max := e.activeConnections.GetBoth()
+	idle := float64(e.pool.NumAllocated())
+
 	// MustNewConstMetric would only panic if there's a label mismatch, which we
 	// have a unit test to cover.
+	ch <- prometheus.MustNewConstMetric(
+		legacyClientPoolActiveConnectionsDesc,
+		prometheus.GaugeValue,
+		float64(active),
+		e.slug,
+		e.slug,
```

Comment:
why is e.slug here twice? same below.

Reply:
we are emitting both the old `thrift_client_name` and new `thrift_pool` labels.